### DISCUSSION TITLE
How to update the minimap

### DISCUSSION THREAD

**Discussion by default-kramer** [default-kramer/HermitsHeresy#17] [General]
Thanks to Sapphire for thinking of this technique:

After doing a massive change, you might want to update the minimap. A convenient way to do this is by putting a layer of block 120 (a Clear Water variant) above the area you edited, or simply above the whole island. Then load your save and replace just one of these water blocks with a solid block and the whole layer will disappear, forcing the minimap to update. Here is a quick proof of concept:

```
#lang racket

(require hermits-heresy
         (submod hermits-heresy undocumented))

(save-dir "C:/Users/kramer/Documents/My Games/DRAGON QUEST BUILDERS II/Steam/76561198073553084/SD/")

(copy-all-save-files! #:from 'B02 #:to 'B00)

(define stage (load-stage 'BT1 'B00))

(define trav
  (traversal
   (cond
     [(< YYY 60)
      (set-block! 'Grassy-Earth)]
     [(= YYY 64)
      (set-block! 120)])))

(time (traverse stage trav))
```

**Comment by default-kramer** [comment]
For large areas, it will take a few minutes to complete. You can watch it progress by opening the map:

![20241226144318_1](https://github.com/user-attachments/assets/16647cb2-c679-4bc0-9557-561976ab7a99)

**Reply by Sapphire645** [reply]
Any ideas for minimap updating new added chunks?

**Reply by default-kramer** [reply]
Not yet, no.
